src/clj/fluree/db/query/api.cljc

```diff
@@ -204,33 +204,33 @@
         (dataset db-map defaults)))))
 
 (defn query-connection-fql
-  [conn query did]
+  [conn query opts]
   (go-try
-   (let [{:keys [t opts] :as query*} (-> query
-                                         syntax/coerce-query
-                                         (update :opts sanitize-query-options did))
-         default-aliases (some-> query* :from util/sequential)
-         named-aliases   (some-> query* :from-named util/sequential)]
-     (if (or (seq default-aliases)
-             (seq named-aliases))
-       (let [ds       (<? (load-dataset conn default-aliases named-aliases t opts))
-             query**  (update query* :opts dissoc :meta :max-fuel ::util/track-fuel?)
-             max-fuel (:max-fuel opts)]
-         (if (::util/track-fuel? opts)
-           (<? (track-query ds max-fuel query**))
-           (<? (fql/query ds query**))))
-       (throw (ex-info "Missing ledger specification in connection query"
-                       {:status 400, :error :db/invalid-query}))))))
-
+    (let [{:keys [t opts] :as sanitized-query} (-> query
+                                                   syntax/coerce-query
+                                                   (update :opts sanitize-query-options (:did opts)))
+          
+          default-aliases (some-> sanitized-query :from util/sequential)
+          named-aliases   (some-> sanitized-query :from-named util/sequential)]
+      (if (or (seq default-aliases)
+              (seq named-aliases))
+        (let [ds            (<? (load-dataset conn default-aliases named-aliases t opts))
+              trimmed-query (update sanitized-query :opts dissoc :meta :max-fuel ::util/track-fuel?)
+              max-fuel      (:max-fuel opts)]
+          (if (::util/track-fuel? opts)
+            (<? (track-query ds max-fuel trimmed-query))
+            (<? (fql/query ds trimmed-query))))
+        (throw (ex-info "Missing ledger specification in connection query"
+                        {:status 400, :error :db/invalid-query}))))))
 
 (defn query-connection-sparql
-  [conn query did]
+  [conn query opts]
   (go-try
-   (let [fql (sparql/->fql query)]
-     (<? (query-connection-fql conn fql did)))))
+    (let [fql (sparql/->fql query)]
+      (<? (query-connection-fql conn fql opts)))))
 
 (defn query-connection
-  [conn query {:keys [format did] :as _opts :or {format :fql}}]
-  (case format
-    :fql (query-connection-fql conn query did)
-    :sparql (query-connection-sparql conn query did)))
+  [conn query {:keys [format] :as opts}]
+  (case (or format :fql)
```

Comment:
Usually we'd handle this in as part of the destructuring syntax (if we were destructuring, which we are here).